### PR TITLE
Allow users to list repositories currently being cloned

### PR DESCRIPTION
**Pull request by mrnugget**
This introduced a new filter to the GraphQL: `cloneStatus`. If set, it
returns only repositories that have the given clone status.

With that we can reduce the number of booleans we use when interacting
with the API: instead of querying for `{cloned: true, notCloned: false}`
(which we did even though the `notCloned` was useless) we can query for
`{cloneStatus: "CLONED"}`.

And now we can also query for `{cloneStatus: "CLONING"}`.

This also changes how the site-admin UI works:
* previously: listing "not cloned" would include "cloning" repositories
* now: listing "not cloned" only lists repositories that are "not cloned".

Subpart of this ticket: https://github.com/sourcegraph/sourcegraph/issues/39328

## Test plan

- unit tests
- manual testing

### CODE REVIEW

File: client/web/src/site-admin/SiteAdminRepositoriesPage.tsx [hunk 1]
```diff
@@ -93,13 +93,19 @@ const FILTERS: FilteredConnectionFilter[] = [
...
                 value: 'cloned',
                 tooltip: 'Show cloned repositories only',
-                args: { cloned: true, notCloned: false },
+                args: { cloneStatus: 'CLONED' },
+            },
+            {
+                label: 'Cloning',
+                value: 'cloning',
+                tooltip: 'Show repositories currently being cloned only',
+                args: { cloneStatus: 'CLONING' },
```
Comment (eseliger) [inline client/web/src/site-admin/SiteAdminRepositoriesPage.tsx:102]: The return of the cloning filter ❤️

File: internal/database/repos.go [hunk 2]
```diff
@@ -1059,6 +1062,10 @@ func (s *repoStore) listSQL(ctx context.Context, tr *trace.Trace, opt ReposListO
 	if opt.OnlyCloned {
 		where = append(where, sqlf.Sprintf("gr.clone_status = 'cloned'"))
```
Comment (eseliger) [inline internal/database/repos.go:1063]: can we refactor these later to use the new prop?

**Review by eseliger — APPROVED** [review]

File: internal/database/repos.go (hunk 2 above)
Comment (mrnugget) [reply internal/database/repos.go:1063]: Yeah, that's the plan: consolidate the booleans in GraphQL into the `CloneStatus` field. And possibly introducing `CloneStatuses` (plural) and `NotCloneStatuses` to cover all cases.

**Comment by mrnugget** [comment]
Hey @sourcegraph/repo-management, I'm gonna go ahead and merge this, since I deem the risk of something breaking really low. As always: happy to address any and all feedback post-merge :)

**Review by ryanslade — COMMENTED** [review]
LGTM 👍

**Pull request merged**